pool/src/pool/submit.rs

```diff
@@ -63,6 +61,84 @@ pub fn execute_submit(
+// a bit to prevent duplicating execute_submit.
+pub fn execute_submit_with_flash_loan(
+    e: &Env,
+    from: &Address,
+    flash_loan: FlashLoan,
+    requests: Vec<Request>,
+    use_allowance: bool,
+) -> Positions {
+    if from == &e.current_contract_address()
+    {
+        panic_with_error!(e, &PoolError::BadRequest);
+    }
+    let mut pool = Pool::load(e);
+
+    // note: check_health is omitted since we always will want to check the health
+    // if a flash loan is involved.
+    let (actions, mut new_from_state, _) =
+        build_actions_from_request(e, &mut pool, from, requests);
+
+    // similarly to usual borrows, we want the flash loan to be debited before
+    // checking the health factor. TODO: We should decide whether the execution ordering
+    // matters here (I'm pretty sure it doesn't but haven't looked much into it).
+    {
+        let mut reserve = pool.load_reserve(e, &flash_loan.asset, true);
+        let d_tokens_minted = reserve.to_d_token_up(flash_loan.amount);
+        new_from_state.add_liabilities(e, &mut reserve, d_tokens_minted);
+        reserve.require_utilization_below_max(e);
+
+        e.events().publish(
+            (
+                Symbol::new(e, "flash_borrow"),
+                &flash_loan.asset,
+                &flash_loan.contract,
+            ),
+            (&flash_loan.amount, d_tokens_minted),
+        );
+    }
+
+    // panics if the new positions set does not meet the health factor requirement
+    // min is 1.0000100 to prevent rounding errors
+    if new_from_state.has_liabilities()
+        && PositionData::calculate_from_positions(e, &mut pool, &new_from_state.positions)
+            .is_hf_under(1_0000100)
+    {
+        panic_with_error!(e, PoolError::InvalidHf);
+    }
+
+    // we deal with the flashloan transfer before the others to allow the flash
+    // loan to yield the repaid or supplied amount in the transfers.
+    TokenClient::new(e, &flash_loan.asset).transfer(&e.current_contract_address(), &flash_loan.contract, &flash_loan.amount); 
+    // calls the receiver contract.
+    FlashLoanClient::new(&e, &flash_loan.contract).exec_op(
+        &e.current_contract_address(),
+        &flash_loan.asset,
+        &flash_loan.amount,
+        &0,
+    );
+
+    // note: at this point, the pool has sum_by_asset(actions.flash_borrow.1) for each involed asset, but the user also has
+    // increased liabilities. These will have to be either fully repaid by now in the requests following the flash borrow
+    // or the user needs to have some previously added collateral to cover the borrow, i.e user is already healthy at this point,
+    // we just have to make sure that they have the balances they are claiming to have through the transfers. 
+
+    if use_allowance {
+        handle_transfer_with_allowance(e, &actions, from, from);
+    } else {
+        handle_transfers(e, &actions, from, from);
+    }
```

Comment:
we should only use allowances here.

When executing transfers with flash loans, the odds the `transfer` requests will be accurate across blocks is unlikely, making simulation difficult.

Reply:
good point, switched on the use allowance flag.